code/mob/living/critter/ai/generic_critter.dm

```diff
@@ -191,7 +191,7 @@
 		var/obj/item/reagent_containers/food/snacks/T = holder.target
 		if(C && T && BOUNDS_DIST(holder.owner, holder.target) == 0)
 			holder.owner.set_dir(get_dir(holder.owner, holder.target))
-			T.Eat(C)
+			T.Eat(C, C, TRUE)
```

Comment:
`by_matter_eater` being set to `TRUE` won't do anything unless the mob has a biolholder and the mattereater effect. See code/obj/item.dm:375

It should almost certainly be `FALSE` in the default critter AI, and overridden by a subclass if you want matter eater abilities on a mob.

Reply:
I thought the true here meant "bypass_utensils" from food_and_drink.dm? To allow critters to eat things such as soup, spaghetti and more?

Reply:
You are absolutely right, I didn't realise `Eat()` overrides don't call parent.